File: leetcode_c_plus_plus/1446.cpp
// 1446. 连续字符
// https://leetcode-cn.com/problems/consecutive-characters/
#include<map>
#include<set>
#include<vector>
#include<cmath>
#include<numeric>
#include<unordered_map>
using namespace std;
#define INT_MAX 2147483637
#define INT_MIN (-INT_MAX - 1)
/**
@File : 1446.cpp
@Time : 2021/12/01 14:26:37
@Author : YuMin Zhang
@State : Indepeedent | Depedent | Half-Depedent
@Thinking :
**/
class Solution {
public:
    int maxPower(string s) {
        int left = 0, right = 0;
        int ans = 0;
        while(right < s.size()) {
            while(right < s.size() && s[left] == s[right]) right++;
            ans = max(ans, right - left);
            left = right;
        }
        return ans;

    }
};

File: leetcode_python/1446.py
# 1446. 连续字符
# https://leetcode-cn.com/problems/consecutive-characters/
from typing import List
from collections import defaultdict
from collections import deque
from math import inf
import bisect
import heapq
'''
@File : 1446.py
@Time : 2021/12/01 14:22:24
@Author : YuMin Zhang
@State : Indepeedent | Depedent | Half-Depedent
@Thinking :
'''
class Solution:
    def maxPower(self, s: str) -> int:
        l, r = 0 , 0
        max_l = 1
        while r < len(s):
            if s[l] == s[r]:
                max_l = max(max_l, r - l + 1)
                r += 1
            else:
                l = r
        return max_l